plugin/trino-delta-lake/src/main/java/io/trino/plugin/deltalake/DeltaLakeMetadata.java

```diff
@@ -2027,9 +1993,9 @@ private void writeCheckpointIfNeeded(ConnectorSession session, SchemaTableName t
     private void cleanupFailedWrite(ConnectorSession session, String tableLocation, List<DataFileInfo> dataFiles)
     {
         try {
-            FileSystem fileSystem = hdfsEnvironment.getFileSystem(new HdfsContext(session), new Path(tableLocation));
+            TrinoFileSystem fileSystem = fileSystemFactory.create(session);
             for (DataFileInfo dataFile : dataFiles) {
```

Comment:
We have now in `TrinoFileSystem` the method `deleteFiles` which performs the deletion in batches on S3. It would probably be a good fit here.

cc @homar